<?xml version="1.0"?>
<launch>
    <arg name="rviz" default="true" />
    <arg name="robot_frame" default="livox_lidar" />
    <arg name="map_frame" default="map" />
    
    <param name="scan_period" type="double" value="0.1" />
    <param name="vertical_angle" type="double" value="25.0" />
    <param name="vertical_angle_resolution" type="double" value="2" />
    <param name="min_vertical_angle" type="double" value="-12.5" /> <!-- minimum angle in the sensor frame -->
    <param name="max_vertical_angle" type="double" value="13." /> <!-- maximum angle in the sensor frame -->
    <param name="horizontal_angle" type="double" value="120.0" />
    <param name="horizontal_angle_resolution" type="double" value="2" />
    <param name="min_horizontal_angle" type="double" value="-60.0" /> <!-- minimum angle in the sensor frame -->
    <param name="max_horizontal_angle" type="double" value="60.0" /> <!-- maximum angle in the sensor frame -->
    <param name="min_dis" type="double" value="2" />
    <param name="max_dis" type="double" value="30.0" />
    <param name="map_resolution" type="double" value="0.1" />
    <param name="optimization_iterations" type="int" value="6" />
    <param name="robot_frame" type="str" value="$(arg robot_frame)"/>
    <param name="map_frame" type="str" value="$(arg map_frame)"/>
    
    <node pkg="ssl_slam" type="ssl_slam_laser_processing_node" name="ssl_slam_laser_processing_node" output="screen">
        <remap from="camera/depth/color/points" to="livox/lidar" />
    </node>
    <node pkg="ssl_slam" type="ssl_slam_odom_estimation_node" name="ssl_slam_odom_estimation_node" output="screen"/>
    <node pkg="ssl_slam" type="ssl_slam_laser_mapping_node" name="ssl_slam_laser_mapping_node" output="screen"/>

    <group if="$(arg rviz)">
        <node launch-prefix="nice" pkg="rviz" type="rviz" name="rviz" args="-d $(find ssl_slam)/rviz/ssl_slam_mapping.rviz" />
    </group>

    <node pkg="hector_trajectory_server" type="hector_trajectory_server" name="trajectory_server_ssl_slam" ns="ssl_slam" >
        <param name="/target_frame_name" value="$(arg map_frame)" />
        <param name="/source_frame_name" value="$(arg robot_frame)" />
        <param name="/trajectory_update_rate" value="10.0" />
        <param name="/trajectory_publish_rate" value="10.0" />
    </node>


</launch>
